tm-monitor/docs/Design-proposal.md

```diff
@@ -0,0 +1,53 @@
+# Design proposal for Tendermint Monitor
+
+Tendermint monitor is designed to be the tool for all related parties to have a close look about the network. 
+
+## Design Goal for monitoring the network
+
+The design goal of Tendermint Monitor is to track **Liveness** & **Safety** of the whole network and certain validator.
+
+ The liveness of the network is in jeopardy if:
+
+* many validators are offline
+* the system barely gather +2/3 votes in recent blocks
+* validators have to wait for many rounds before commiting new block
+
+
+The safety of the network is in jeopardy if:
+
+* There exists a monopoly of Atom token, +1/3
+
+* Not enough Atoms are bonded
+
+* Certain validator attracts too many delegation
+
+
+## Design Goal for monitoring a validator
+
+The validator must monitor its **availability** and **safety** closely.  There are some early-stage indicators to be noticed:
+
+*  Are most of previous precommits included in blocks?
+* How many proposals it missed before?
+
+
+The safety of a validator is under pressure if :
+
+* it's not connected with enough persistent peers 
+* there is not enought peers in its address book
+
+## Tendermint Monitor Atchitecture
+
+The architecture of new version of monitor will be like this:
+
+![WechatIMG114](https://github.com/bianjieai/tendermint-tools/raw/bianjie/tm-monitor/tm-monitor/docs/pics/WechatIMG114.jpeg)
+
+The data will be stored in Mongo to facilitate better querying.
```

Comment:
better querying? did you compare it to other popular options like https://clickhouse.yandex/ or elasticsearch or influxdb?

Reply:
I agree here. I think the most important point of tm-monitor is to map out measurement points for the user (especially ones that the API doesn't show very well). Most of the aggregating, storing and presenting those values should be left to software that is already written. To go back to the ethtool page example that Anton mentioned, if the data is available in InfluxDB, I can easily create a graph like that using Grafana. Maybe we should even include a few templates like that for easy setup.

Reply:
In short: the dashboards and MonoDB should not be in scope for the tm-monitor application, only examples on how to use the software. Also, let's include Grafana and ELK visualization templates. Do not write our own dashboard JavaScript, it's reinventing the wheel.